src/main/java/seedu/duke/ui/TextUi.java

```diff
@@ -0,0 +1,82 @@
+package seedu.duke.ui;
+
+import java.nio.BufferOverflowException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+/**
+ * TextUI class for reading user inputs and printing outputs.<br/>
+ * Subclasses of TextUI class can override separator, printErrorFor and printOutputMessage.<br/>
```

Comment:
I think you may need to include a setter that accepts a string `length == 1` to properly override the separator.

The private variable cannot be accessed by subclasses of TextUi

Reply:
Okay I'll write a setter. 
But can have length>1 right if developers have a pattern in mind to use?

Reply:
Yes, up to the implementer's discretion!
Length check is if we really want to be strict with the output format length